Add Restock Gigantor cfg

lpgagnon opened the pull request:
No model import, just use model from restock if installed. Based on
NFS configs.
Added to Folding panels, for lack of a better one

Per discord discussion, it may be better to create a new part for "station" panels; but that's a bigger job than I'm prepared for, so sharing what I have, might at least serve as starting point.

Probably shouldn't merge this if it's going to get removed later; that'd presumably be craft-breaking.

lpgagnon committed:
Add Restock Gigantor cfg

No model import, just use model from restock if installed. Based on
NFS configs.
Added to Folding panels, for lack of a better one

DRVeyl commented on GameData/ROSolar/Data/Models/ModelData-Restock.cfg:
```diff
@@ -0,0 +1,18 @@
+	name = ReStock_gigantor
+	title = ReStock Gigantor
+	modelName = ReStock/Assets/Electrical/restock-solarpanel-gigantor
+//	panelLength = 6
+//	panelWidth = 2.2
+	panelArea = 11.4 // 1.9*6, there's a gap.
+	secondaryTransformName = suncatcher
+	pivotName = suncatcher // FIXME probably shouldn't be same as secondaryTransformName, but seems to work
```
Ideally you would put the overall model length and width in, to support independent scaling instead of scaling the entire model.  I believe ROSolar only does independent scaling for the static panels, but eh.

pivotName and secondaryTransformName should usually be the same as what the original part maker provides in their cfg.  Deviating should be intentional.  pivotName controls where/how the panel can rotate, secondaryTransformName controls how the raycasting & suntracking work, and its transform's forward direction needs to be aligned to the panel facing (surface normal)

lpgagnon replied:
Length & width: can do, I had no idea which was preferred. Need to set lengthWidth=true?
How to deal with the fact that the full width isn't covered in panel? give the actual panel width or the model width?

There's no pivotName in the restock gigantor cfg; no transform specified at all other than secondaryTransformName.

DRVeyl replied:
Don't set `lengthWidth`=true unless the panel wants to independently scale its length and width, instead of overall scaling.  Most parts except static panels scale the entire thing consistently.  `panelArea` is the critical one to set.

I do not think it actually matters how you specify the length and width [solar-receiving section vs total model dimension].  When in length-width mode, `panelArea` will be scaled by the [input value]/[model definition value that you specify here] for each dimension.  For consistency, I believe we have used the total model extents, but I am definitely not certain at this point.

The default `pivotName` is "sunPivot" If the original .cfg did not specify, it should not be needed here, either.  We should default sanely.

lpgagnon committed:
Do specify length & width; don't specify pivotName

lpgagnon committed:
Add rotationOffset

so parts placed in symmetry get a more useful orientation

lpgagnon committed:
Fix 'add CORE if missing' selector

Original broken version was copied from NFS file, which is being fixed separately
